### ISSUE TITLE
gulpfile.babel.js compatibility issuez

### ISSUE THREAD

**Issue by jmeas** [rollup/rollup#649]
tl;dr Rollup **is** compatible with `gulpfile.babel.js`, but it doesn't seem ideal.

My projects use `.babelrc` to configure Babel. This works great. This also lets me use `gulpfile.babel.js` and author my Gulpfile in ES2015.

One thing I've run into is that the Gulpfile needs the full `es2015` Babel plugin in order for imports and exports to be resolved. However, my library code needs Rollup's special es2015 plugin, which omits modules, instead.

The only way I can think of to combine these two requirements is for `.babelrc` to describe **just** the Babel configuration for the Gulpfile, and then specify the library's Babel config within the Gulpfile itself. This doesn't seem ideal, because I would expect / prefer `.babelrc` to refer to the code of the lib itself.

I don't think that this is necessarily an issue with Rollup. I'm more just curious to hear if anyone else has figured out a clever solution I can't think of :)

**Comment by jmeas** [comment]
Closing because I don't think this is an issue anymore. I realized that because the latest Node supports so many ES2015 features, it doesn't really make as much sense to use `gulpfile.babel.js` anymore. The only difference that you really get is the module syntax, which doesn't seem worth the hassle.

So my fix for this issue is just not using `gulpfile.babel.js` anymore :v:

**Comment by Rich-Harris** [comment]
Nice! For anyone else who lands here, I've solved similar problems in the past by having a `.babelrc` at the root that specifies plugins needed by things like test files or gulpfiles (i.e. `es2015`), and a separate `.babelrc` in my `src` dir that specifies the plugins that are relevant to the code itself (i.e. `es2015-rollup`). Babel will pick the nearest `.babelrc` to the file being compiled.

**Comment by jmeas** [comment]
That's a great idea as well, @Rich-Harris :v:
